src/query/storage/m3/cluster_resolver.go

```diff
@@ -301,23 +302,35 @@ func aggregatedNamespaces(
 func resolveClusterNamespacesForQueryWithRestrictFetchOptions(
 	now, start time.Time,
 	clusters Clusters,
-	restrict *storage.RestrictFetchOptions,
+	fetchOptions *storage.RestrictFetchOptions,
 ) (queryFanoutType, ClusterNamespaces, error) {
 	coversRangeFilter := newCoversRangeFilter(coversRangeFilterOptions{
 		now:        now,
 		queryStart: start,
 	})
+
+	if fetchOptions == nil {
```

Comment:
Will restrict fetch options always be not nil if no headers were set? 

Currently we don't set it to a non-nil value unless one of the headers were set:
https://github.com/m3db/m3/blob/f6640a1a287a86c611f9f14fdd62cf9d73096c45/src/query/api/v1/handler/fetch_options.go#L147

Reply:
Discussed offline.